octobot/limits.py

```diff
@@ -0,0 +1,73 @@
+
+
+def _apply_exchanges_limits(dict_config, logger, limit):
+    exchanges = [
+        f"{exchange}[{config.get(common_constants.CONFIG_EXCHANGE_TYPE, common_constants.CONFIG_EXCHANGE_SPOT)}]"
+        for exchange, config in dict_config[common_constants.CONFIG_EXCHANGES].items()
+        if config.get(common_constants.CONFIG_ENABLED_OPTION, True)
+    ]
+    if limit != constants.UNLIMITED_ALLOWED and len(exchanges) > limit:
+        enabled_exchanges = []
+        for exchange, config in dict_config[common_constants.CONFIG_EXCHANGES].items():
+            if config.get(common_constants.CONFIG_ENABLED_OPTION, True):
+                if len(enabled_exchanges) < limit:
+                    enabled_exchanges.append(exchange)
+                else:
+                    config[common_constants.CONFIG_ENABLED_OPTION] = False
+                    logger.warning("Disabled : " + exchange)
+        logger.error(f"Too many enabled exchanges, maximum allowed is {limit}. Enabled : {enabled_exchanges}")
+
+
+def _apply_symbols_limits(dict_config, logger, limit):
+    if limit != constants.UNLIMITED_ALLOWED:
+        enabled_symbols = []
+        for currency, crypto_currency_data in dict_config[common_constants.CONFIG_CRYPTO_CURRENCIES].items():
+            if crypto_currency_data.get(common_constants.CONFIG_ENABLED_OPTION, True):
+                if len(enabled_symbols) >= limit:
+                    crypto_currency_data[common_constants.CONFIG_ENABLED_OPTION] = False
+                    logger.warning(f"Disabled : {currency}")
+                    continue
+                updated_symbols = []
+                for symbol in crypto_currency_data[common_constants.CONFIG_CRYPTO_PAIRS]:
+                    if symbol == common_constants.CONFIG_SYMBOLS_WILDCARD[0] \
+                            or symbol == common_constants.CONFIG_SYMBOLS_WILDCARD:
+                        crypto_currency_data[common_constants.CONFIG_ENABLED_OPTION] = False
+                        logger.warning(f"Disabled wildcard symbol for {currency}")
+                        break
+                    else:
+                        if len(enabled_symbols) < limit:
+                            enabled_symbols.append(symbol)
+                            updated_symbols.append(symbol)
+                        else:
+                            logger.warning(f"Disabled : {symbol}")
+                crypto_currency_data[common_constants.CONFIG_CRYPTO_PAIRS] = updated_symbols
+        if len(trading_api.get_config_symbols(dict_config, True)) > limit:
+            logger.error(f"Too many trading pairs, maximum allowed is {limit}. Enabled : {enabled_symbols}")
+
+
+def apply_config_limits(configuration):
+    logger = logging.get_logger("ConfigurationLimits")
+    try:
+        _apply_exchanges_limits(configuration.config, logger, constants.MAX_ALLOWED_EXCHANGES)
+        _apply_symbols_limits(configuration.config, logger, constants.MAX_ALLOWED_SYMBOLS)
```

Comment:
Shouldn't we check here if we have to call the apply methods. For example by adding:
```
if constants.MAX_ALLOWED_EXCHANGES != constants.UNLIMITED_ALLOWED
```
before the first method